crates/subspace-networking/src/utils.rs

```diff
@@ -163,7 +163,7 @@ impl SemState {
         if let Some(dec) = self.capacity.checked_sub(delta) {
             self.capacity = dec;
         } else {
-            panic!("SemState::shrink(): invalid shrink, state = {self:?}");
+            warn!(state = ?self, "SemState::shrink(): invalid shrink");
```

Comment:
The right invariant to enforce would be: we capacity never falls below the initial capacity (i.e) user doesn't shrink() more than the expand(). 

I would prefer: either leave as is, or enforce the above mentioned invariant and fail the shrink() with proper error code.  Just doing warn() and return is not ideal

Reply:
As is meaning panicking?

Reply:
Sorry no, agree with you, no panic. We should just return error